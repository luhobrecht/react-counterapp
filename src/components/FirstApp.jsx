import { Fragment } from "react"
import PropTypes from "prop-types"

export  function FirstApp ({title, subTitle,name}) {
    return (
    <Fragment>  
    <h1 data-testid="test-title">{title}</h1>
    <p>{subTitle}</p>
    <p>{subTitle}</p>
    <p>{name}</p>
    </Fragment>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

FirstApp.defaultProps = {
    name: "Lucía Hobrecht",
    subTitle: 'No hay subtítulo',
    title: 'No hay título'
}