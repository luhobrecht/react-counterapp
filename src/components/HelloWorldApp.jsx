function HelloWorldApp() {
    // Es lo mismo que hacer document.createElementBy.....
    return <h1>Hola Mundo</h1>;
}

export default HelloWorldApp;