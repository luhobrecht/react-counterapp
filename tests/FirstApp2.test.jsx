import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/components/FirstApp"

 describe('Prueba sobre el component FirstApp', () => { 
     const title = 'Hola, Soy Goku'
     const subTitle= 'Soy un subtitulo'
    test('should make match with the snapshot', () => { 
        
      const {container} = render( <FirstApp title= { title } /> )

      expect (container).toMatchSnapshot();
     }),

     test('should show the message "Hola, Soy Goku"', () => { 
        
      render( <FirstApp title= {title} /> );
      expect( screen.getAllByText(title) ).toBeTruthy();

      }),

      test('should show title in h1', () => { 
         
         render( <FirstApp title= {title} /> );
         expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);

       });

       test('should show the subtitle send by props', () => { 
          
         render( <FirstApp 
            title= {title} 
            subTitle= {subTitle}
            /> );
         expect(screen.getAllByText(subTitle).length).toBe(2)

        })

  })