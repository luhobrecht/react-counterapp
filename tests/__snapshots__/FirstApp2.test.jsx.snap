// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre el component FirstApp should make match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Lucía Hobrecht
  </p>
</div>
`;
